toml/decoder.py

```diff
@@ -702,17 +703,18 @@ def load_value(self, v, strictly_valid=True):
             elif v[0] == '+':
                 v = v[1:]
             v = v.replace('_', '')
-            if '.' in v or 'e' in v or 'E' in v:
-                if '.' in v and v.split('.', 1)[1] == '':
-                    raise ValueError("This float is missing digits after "
-                                     "the point")
-                if v[0] not in '0123456789':
-                    raise ValueError("This float doesn't have a leading "
-                                     "digit")
-                v = float(v)
-                itype = "float"
-            else:
-                v = int(v)
+            if 'x' not in v:
```

Comment:
Why 'x' and not 'o' or 'b' ?

Reply:
Next line is 
```if '.' in v or 'e' in v or 'E' in v:```
so hex numbers containing 'e' will be mistaken as a float. That is not a problem with binary and octal numbers so only 'x' is needed there